services/blockstorage/internodesync/state_processing_blocks.go

```diff
@@ -84,9 +98,78 @@ func (s *processingBlocksState) processState(ctx context.Context) syncState {
 		}
 	}
 
-	if !s.factory.conduit.drainAndCheckForShutdown(ctx) {
+	if !s.conduit.drainAndCheckForShutdown(ctx) {
 		return nil
 	}
 
 	return s.factory.CreateCollectingAvailabilityResponseState()
 }
+
+func (s *processingBlocksState) validatePosChain(ctx context.Context, blocks []*protocol.BlockPairContainer, blockSyncReferenceMaxDistance time.Duration, receivedSyncBlocksOrder gossipmessages.SyncBlocksOrder) error {
+	if receivedSyncBlocksOrder == gossipmessages.SYNC_BLOCKS_ORDER_RESERVED && s.syncBlocksOrder == gossipmessages.SYNC_BLOCKS_ORDER_ASCENDING {
+		return nil
+
+	} else if s.syncBlocksOrder != receivedSyncBlocksOrder {
+		return errors.New("received chunk with blocks order which does not match blockSync expected blocks order")
+
+	} else if receivedSyncBlocksOrder == gossipmessages.SYNC_BLOCKS_ORDER_DESCENDING {
+		firstBlock := blocks[0]
+		if nextBlock, _ := s.storage.GetBlock(firstBlock.TransactionsBlock.Header.BlockHeight() + 1); nextBlock != nil { // will verify hash pointer to block
+			// prepend
+			blocks = append([]*protocol.BlockPairContainer{nextBlock}, blocks...)
+		} else {
+			ref, err := s.management.GetCurrentReference(ctx, &services.GetCurrentReferenceInput{})
+			if err != nil {
+				s.logger.Error("management.GetCurrentReference should not return error", log.Error(err))
+				return err
+			}
+			if firstBlock.TransactionsBlock.Header.ReferenceTime() + primitives.TimestampSeconds(blockSyncReferenceMaxDistance/time.Second) < ref.CurrentReference {
```

Comment:
didn't you say we will remove this ? if node is honest then it will be ok (or check vs block time instead? )
if you remove this then remember to remove managemnt service from all the tree of dependancies